src/MoviePage/index.js

```diff
@@ -29,6 +23,11 @@ function MoviePage() {
     }
   }
 
+  async function setMoviesData() {
+    const result = await getMovies(page);
+    setMovies(movies.concat(result.results));
+  }
+
   useEffect(() => {
     window.addEventListener("scroll", onScrollDepth);
```

Comment:
this event is attached after every render, you need to pass an empty array as a dependency, that way it will just get executed once after the first render, also you need to return a cleanup function with the removeListener for this event to avoid memory leaks

Reply:
Miss this. Thanks!!

Reply:
I guess you meant componentWillUnmount, but with the empty dependencies array I think it runs like componentDidMount and will run just once, we can debug this later on, add a //TODO comment

Reply:
Yes `componentWillUnmount` 😅

Reply:
Fix the issue!!!
The problem wasn't on the `useEffect` was the `onScrollDepth` function. Since we are working updating the `previews` state of `page` we need to send an `array` so we can properly update that state and that fix the issue
@jsainthilaire @mjimenez07

Reply:
Nice catch, just tested and looks great 🍻